packages/gatsby/src/schema/schema.js

```diff
@@ -176,28 +182,35 @@ const addThirdPartySchemas = ({
   })
 }
 
-// const addCustomResolveFunctions = ({
-//   schemaComposer,
-//   resolvers,
-//   parentSpan,
-// }) => {
-//   Object.entries(resolvers).forEach(([typeName, fields]) => {
-//     if (schemaComposer.has(typeName)) {
-//       const tc = schemaComposer.getTC(typeName)
-//       Object.entries(fields).forEach(([fieldName, resolve]) => {
-//         if (tc.hasField(fieldName)) {
-//           const resolver = tc.getField(fieldName).resolve
-//           tc.extendField(fieldName, {
-//             resolve,
-//           })
-//         } else {
-//           const fieldConfig = resolve
-//           tc.addFields({ [fieldName]: fieldConfig })
-//         }
-//       })
-//     }
-//   })
-// }
+const addCustomResolveFunctions = async ({
+  schemaComposer,
+  resolvers,
+  parentSpan,
+}) => {
+  const intermediateSchema = schemaComposer.buildSchema()
+  const addResolvers = resolvers => {
+    Object.keys(resolvers).forEach(typeName => {
+      const fields = resolvers[typeName]
+      if (schemaComposer.has(typeName)) {
+        const tc = schemaComposer.getTC(typeName)
+        Object.keys(fields).forEach(fieldName => {
+          const fieldConfig = fields[fieldName]
+          if (tc.hasField(fieldName)) {
+            tc.extendField(fieldName, fieldConfig)
```

Comment:
When extending a field, how do we expect people to get to the original field resolver (e.g. when they just want to wrap it). Should they get it from the intermediateSchema?

Reply:
One other thing to think about is if we allow changing field types, and adding new fields, they are not automatically reflected in the InputObjectType

Reply:
We probably should only allow changing resolver in case field is defined, that is a good point.